runtime/drivers/duckdb/olap.go

```diff
@@ -181,6 +186,344 @@ func (c *connection) EstimateSize() (int64, bool) {
 	return fileSize(paths), true
 }
 
+// AddTableColumn implements drivers.OLAPStore.
+func (c *connection) AddTableColumn(ctx context.Context, tableName, columnName, typ string) error {
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:    fmt.Sprintf("ALTER TABLE %s ADD COLUMN %s %s", safeSQLName(tableName), safeSQLName(columnName), typ),
+			Priority: 1,
+		})
+	}
+
+	version, exist, err := c.tableVersion(tableName)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		return fmt.Errorf("table %q does not exist", tableName)
+	}
+	dbName := dbName(tableName, version)
+	// ignore query cancellations
+	ctx = context.WithValue(context.Background(), connCtxKey{}, connFromContext(ctx))
+	err = c.Exec(ctx, &drivers.Statement{
```

Comment:
If the first query(Alter Table) succeeds then the view is no longer queryable so we need second query to succeed as well. Although the query can still fail due to any number of reasons but most probable because of query cancellations.

Reply:
Okay, then it should only use a non-cancellable ctx after the first query

Reply:
yeah that should be done.